src/common/io/src/buffer/buffer_read_datetime_ext.rs

```diff
@@ -97,15 +97,30 @@ where R: BufferRead
             loop {
                 buf.clear();
                 let size = self.keep_read(&mut buf, |f| (b'0'..=b'9').contains(&f))?;
-                match get_time(&mut buf, size) {
-                    Ok(time) => times.push(time),
-                    Err(e) => return Err(e),
-                }
-                if times.len() == 3 {
+                if size == 0 {
                     break;
+                } else {
+                    match get_time(&mut buf, size) {
```

Comment:
let time = get_time()?;
times.push(time);

Reply:
Got and there has a clippy err

```
error: it looks like the same item is being pushed into this Vec
   --> src/common/io/src/buffer/buffer_read_datetime_ext.rs:116:21
    |
116 |                     times.push(0);
    |                     ^^^^^
    |
    = note: `-D clippy::same-item-push` implied by `-D warnings`
    = help: try using vec![0;SIZE] or times.resize(NEW_SIZE, 0)
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#same_item_push

```